### PR TITLE
test: remove isHeadless check

### LINKED ISSUE #12013: Move conditional "skip" in "headful.spec" to TestExpectations.json
The tests in the mentioned file are conditionally skipped, which is not reflected in the TestExpectations.json file:

https://github.com/puppeteer/puppeteer/blob/1250d25e5c20badd3cbb6c8e465aa70c8dabea07/test/src/headful.spec.ts#L19

This causes sometimes issues in our CI.

CC @OrKoN

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Closes #12013

### CODE REVIEW

**Review by OrKoN — APPROVED** [review]

File: test/src/headful.spec.ts [hunk 1]
```diff
@@ -12,11 +12,11 @@ import expect from 'expect';
 import type {PuppeteerLaunchOptions} from 'puppeteer-core/internal/node/PuppeteerNode.js';
 import {rmSync} from 'puppeteer-core/internal/node/util/fs.js';
 
-import {getTestState, isHeadless, launch} from './mocha-utils.js';
+import {getTestState, launch} from './mocha-utils.js';
 
 const TMP_FOLDER = path.join(os.tmpdir(), 'pptr_tmp_folder-');
 
-(!isHeadless ? describe : describe.skip)('headful tests', function () {
+describe('Headful only tests', function () {
```
Comment (OrKoN) [inline test/src/headful.spec.ts]: let's not change the test name

File: test/TestExpectations.json [hunk 2]
```diff
@@ -47,6 +47,12 @@
     "parameters": ["webDriverBiDi"],
     "expectations": ["SKIP"]
   },
+  {
+    "testIdPattern": "[headful.spec] *",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["firefox"],
+    "expectations": ["SKIP"]
```
Comment (whimboo) [inline test/TestExpectations.json:54]: @Lightning00Blade Hm, what was the reason to skip all the tests for Firefox?

Comment (Lightning00Blade) [reply test/TestExpectations.json:54]: Previously the was skipped by the check, now we skip it from the expectations. 
But I see that it passes locally will send a PR to enable it.

Comment (Lightning00Blade) [reply test/TestExpectations.json:54]: https://github.com/puppeteer/puppeteer/pull/12037

Comment (whimboo) [reply test/TestExpectations.json:54]: Thanks!

**Pull request merged**